op-bindings/bindgen/generator_remote.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/etherscan"
+)
+
+type bindGenGeneratorRemote struct {
+	bindGenGeneratorBase
+	contractDataClients map[string]contractDataClient
+	tempArtifactsDir    string
+}
+
+type contractDataClient interface {
+	FetchAbi(address string) (string, error)
+	FetchDeployedBytecode(address string) (string, error)
+	FetchDeploymentTxHash(address string) (string, error)
+	FetchDeploymentTx(txHash string) (etherscan.TxInfo, error)
+}
+
+type remoteContract struct {
+	Name            string            `json:"name"`
+	Verified        bool              `json:"verified"`
+	Deployments     map[string]string `json:"deployments"`
+	DeploymentSalt  string            `json:"deploymentSalt"`
+	DeployerAddress string            `json:"deployerAddress"`
+	Abi             string            `json:"abi"`
+	InitBytecode    string            `json:"initBytecode"`
```

Comment:
Perhaps this should be a `hexutil.Bytes`

Reply:
What would be the reason? The bytecode is provided by the user (via `artifacts.json`) as a string for readability, and almost everywhere these values are used as strings for comparison against the string response of Etherscan API. Only once are they converted to byte arrays which is when they're written to a temp file for abigen to use